utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/elasticsearch/ElasticQuery.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2015-2017 Hewlett Packard Enterprise Development LP.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.utilities.familytasksubmitter.elasticsearch;
+
+import com.hpe.caf.api.ConfigurationException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.SearchType;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.queryStringQuery;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author mcgreeva
+ */
+public class ElasticQuery
+{
+    private final static Logger LOG = LoggerFactory.getLogger(ElasticQuery.class);
+
+    //Refer to https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-from-size.html
+    private final static int ELASTIC_INDEX_MAX_RESULT_WINDOW = 10000;
+
+    private final String ELASTIC_CLUSTER_NAME = getEnvironmentValue("CAF_ELASTIC_CLUSTERNAME", "docker-cluster");
+    private final String ELASTIC_INDEX_NAME = getEnvironmentValue("CAF_ELASTIC_INDEX_NAME", "data-processing");
+    private final String ELASTIC_TYPE = getEnvironmentValue("CAF_ELASTIC_TYPE", "document");
+    private final String ELASTIC_HOST_NAME = getEnvironmentValue("CAF_ELASTIC_HOSTNAME", "vagrant-mesos");
+    private final int ELASTIC_PORT = Integer.parseInt(getEnvironmentValue("CAF_ELASTIC_TRANSPORT_PORT", "9300"));
+    private final TransportClient transportClient;
+    private final String ROOT_REFERENCE;
+
+    public ElasticQuery(final String ROOT_REFERENCE) throws ConfigurationException
+    {
+        this.transportClient = getTransportClient();
+        this.ROOT_REFERENCE = ROOT_REFERENCE;
+    }
+
+    public JSONObject createFamilyJson(final String filename)
```

Comment:
Why do you not just convert the hits and it's fields to Map<String,Multimap<String, String>>? You have fields and references...